src/xdsl/parser.py

```diff
@@ -349,6 +362,18 @@ def parse_string(self,
             return True
         raise ParserError(self._pos, f"'{contents}' expected")
 
+    def parse_optional_string(self,
+                              contents: str,
+                              skip_white_space: bool = True) -> bool | None:
```

Comment:
Why return True or None? Doesn't it make more sense to return True or False? The usage should be the same, right?

Reply:
All functions parsing something optionally return a None in case nothing was parsed. This is because we have "combinator" functions (like parse_list) that checks if the result is None for instance.